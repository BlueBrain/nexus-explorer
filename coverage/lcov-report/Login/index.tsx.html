<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Login/index.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Login</a> index.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">90.91% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as React from 'react';
import { Card, Dropdown, Icon, Menu } from 'antd';
&nbsp;
import './Login.less';
&nbsp;
const logo = require('../../logo.svg');
&nbsp;
export type Realm = {
  name: string;
  authorizationEndpoint: string;
};
&nbsp;
export interface LoginProps {
  realms: Realm[];
  clientId: string;
  hostName: string;
  busy?: boolean;
}
&nbsp;
const Login: React.FunctionComponent&lt;LoginProps&gt; = ({
  realms,
  clientId,
  hostName,
}) =&gt; {
  const [realm, setRealm] = React.useState(realms[0]);
&nbsp;
  const menu = (
    &lt;Menu
      onClick={<span class="fstat-no" title="function not covered" >({</span> key }) =&gt; <span class="cstat-no" title="statement not covered" >setRealm(realms.filter(<span class="fstat-no" title="function not covered" >r </span>=&gt; <span class="cstat-no" title="statement not covered" >r.name === key)</span>[0])}</span>
    &gt;
      {realms.map(realm =&gt; (
        &lt;Menu.Item key={realm.name}&gt;{realm.name}&lt;/Menu.Item&gt;
      ))}
    &lt;/Menu&gt;
  );
&nbsp;
  return (
    &lt;div className="Login"&gt;
      &lt;Card
        cover={&lt;img className="logo" alt="Nexus logo" src={logo} /&gt;}
        actions={[
          &lt;a
            className="link"
            key="login"
            href={`${
              realm.authorizationEndpoint
            }?client_id=${clientId}&amp;response_type=token&amp;scope=openid&amp;nonce=123456&amp;redirect_uri=${hostName}/authRedirect`}
          &gt;
            {realms.length === 1 ? (
              'Log in '
            ) : (
              &lt;React.Fragment&gt;
                Log in with{' '}
                &lt;Dropdown overlay={menu}&gt;
                  &lt;span className="realm"&gt;{realm.name}&lt;/span&gt;
                &lt;/Dropdown&gt;{' '}
              &lt;/React.Fragment&gt;
            )}
            &lt;Icon type="login" /&gt;
          &lt;/a&gt;,
        ]}
      &gt;
        &lt;p className="message"&gt;Please log in to continue.&lt;/p&gt;
      &lt;/Card&gt;
    &lt;/div&gt;
  );
};
&nbsp;
export default Login;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 16 2019 08:49:36 GMT+0100 (CET)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
