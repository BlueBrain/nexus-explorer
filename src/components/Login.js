import React, { Fragment } from "react";
import PropTypes from "prop-types";
import styled from "styled-jss";
import { Auth, WithStore } from "@bbp/nexus-react";
import { auth } from "../store/actions";
import CopyToClipboard from "./CopyToClipboard";

const CopyToken = styled("small")({
  fontSize: "12px",
  backgroundColor: "white",
  padding: "1em",
  borderRadius: "2px",
  "& a": {
    color: "#888",
    transition: "color 200ms"
  },
  "& a.copied": {
    color: "rgba(185, 233, 212, 1)"
  }
});

const CopyTokenContainer = ({ copied }) => (
  <CopyToken>
    <a
      className={copied ? "copied" : ""}
      target="_blank"
      rel="noopener noreferrer"
    >
      {" "}
      copy token{" "}
    </a>
  </CopyToken>
);

CopyTokenContainer.propTypes = {
  copied: PropTypes.bool
};

const Login = () => (
  <WithStore
    mapStateToProps={({ auth, config }) => ({
      name: auth.tokenOwner,
      token: auth.token,
      loginURI: config.loginURI
    })}
    mapDispatchToProps={{
      authenticate: auth.authenticate,
      logout: auth.logout
    }}
  >
    {({ name, loginURI, authenticate, logout, token }) => (
      <Fragment>
        {token && (
          <CopyToClipboard text={"copy token to clipboard"} value={token}>
            <CopyTokenContainer />
          </CopyToClipboard>
        )}
        <div className="login-block">
          <Auth.Container
            name={name}
            authenticate={() => authenticate(window.location.href)}
            logout={logout}
            loginURI={loginURI}
          />
        </div>
      </Fragment>
    )}
  </WithStore>
);

export default Login;
