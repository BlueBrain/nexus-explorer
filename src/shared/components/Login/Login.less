@import '../../lib.less';

.Login {
  background-color: #fff;
}
