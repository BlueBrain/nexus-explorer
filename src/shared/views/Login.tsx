import * as React from 'react';
import { connect } from 'react-redux';
import { Realm } from '@bbp/nexus-sdk';
import LoginBox from '../components/Login';
import { push } from 'connected-react-router';
import getUserManager from '../../client/userManager';
import { RootState } from '../store/reducers';
import { UserManager } from 'oidc-client';
import * as configActions from '../store/actions/config';
import { notification } from 'antd';

export interface LoginViewProps {
  realms: Realm[];
  redirect(): void;
  setPreferredRealm(name: string): void;
  preferredRealm?: string;
  userManager?: UserManager;
}

const Login: React.FunctionComponent<LoginViewProps> = props => {
  const { realms, redirect } = props;
  const defaultRealm: Realm =
    realms.find(r => r.label === props.preferredRealm) || props.realms[0];

  const [preferredRealm, setPreferredRealm] = React.useState(defaultRealm.name);

  if (realms.length === 0 || !realms) {
    redirect();
    return null;
  }
  return (
    <LoginBox
      realms={realms.map(r => r.name)}
      selectedRealm={preferredRealm}
      onLogin={async (e: React.SyntheticEvent) => {
        try {
          e.preventDefault();
          props.setPreferredRealm(preferredRealm);
          props.userManager && (await props.userManager.signinRedirect());
        } catch (error) {
          switch (error.message) {
            case 'Network Error':
              notification.error({
                message: 'We could not log you in',
                description: (
                  <div>
                    <p>
                      Nexus Web could not connect to the openId provider
                      configured for this instance.
                    </p>{' '}
                    <p>Please contact your system administrators.</p>
                  </div>
                ),
                duration: 0,
              });
              break;
            default:
              notification.error({
                message: 'We could not log you in',
                description: (
                  <div>
                    <p>An unknown problem occured.</p>{' '}
                    <p>Please contact your system administrators.</p>
                  </div>
                ),
                duration: 0,
              });
              break;
          }
        }
      }}
      onRealmSelected={(name: string) => setPreferredRealm(name)}
    />
  );
};

const mapStateToProps = (state: RootState) => {
  const { auth, config } = state;
  const userManager = getUserManager(state);
  return {
    userManager,
    realms:
      (auth.realms &&
        auth.realms.data &&
        auth.realms.data.results &&
        auth.realms.data.results.filter(
          r => r.label !== 'serviceaccounts' && !r.deprecated
        )) ||
      [],
    preferredRealm: config.preferredRealm || undefined,
  };
};

const mapDispatchToProps = (dispatch: any) => ({
  redirect: () => dispatch(push('/')),
  setPreferredRealm: (name: string) =>
    dispatch(configActions.setPreferredRealm(name)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Login);
