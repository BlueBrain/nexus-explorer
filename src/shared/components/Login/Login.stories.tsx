import React = require('react');
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Login from './index';

storiesOf('Components/Login', module).add(
  'Login',
  withInfo(`
    The Login component allows to redirect to the Auth Provider page.

    ~~~js
      <Login loginURL="https://keycloak.org?redirect=https://bbp.epfl.ch/nexus/" />
    ~~~
  `)(() => (
    <div style={{ margin: '50px 40px 0px' }}>
      <Login loginURL="" />
    </div>
  ))
);
