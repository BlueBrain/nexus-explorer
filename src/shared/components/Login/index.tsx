import * as React from 'react';
import { Card, Icon } from 'antd';

import './Login.less';

const logo = require('../../logo.svg');

export interface LoginProps {
  loginURL: string;
}

const Login: React.SFC<LoginProps> = ({ loginURL }) => (
  <div className="Login">
    <Card
      cover={<img className="logo" alt="Nexus logo" src={logo} />}
      actions={[
        <a key="login" href={loginURL}>
          Login <Icon type="login" />
        </a>,
      ]}
    >
      <p className="message">please login to continue.</p>
    </Card>
  </div>
);

export default Login;
