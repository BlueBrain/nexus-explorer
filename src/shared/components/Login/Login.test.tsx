import React = require('react');
import { shallow, mount } from 'enzyme';

import Login from './index';

const shallowLogin = shallow(<Login loginURL="https://keycloack.com" />);
const fullDOMLogin = mount(<Login loginURL="https://keycloack.com" />);

describe('login component', () => {
  it('should render correctly', () => {
    expect(shallowLogin).toMatchSnapshot();
  });

  it('should have an anchor tag with correct href attribute', () => {
    expect(fullDOMLogin.find('a')).toBeTruthy();
    expect(
      fullDOMLogin
        .find('a')
        .getDOMNode()
        .getAttribute('href')
    ).toEqual('https://keycloack.com');
  });
});
