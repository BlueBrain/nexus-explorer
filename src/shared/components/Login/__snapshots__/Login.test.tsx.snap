// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login component should render correctly 1`] = `
<div
  className="Login"
>
  <Card
    actions={
      Array [
        <a
          href="https://keycloack.org"
        >
          Login 
          <Icon
            type="login"
          />
        </a>,
      ]
    }
    cover={
      <img
        alt="Nexus logo"
        className="logo"
        src="/mock/path"
      />
    }
  >
    <p
      className="message"
    >
      please login to continue.
    </p>
  </Card>
</div>
`;
